### PR TITLE
[DOC] Add encoding external/internal example to encoding.rdoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Adapted from an example in encoding.c.

### CODE REVIEW

File: doc/encoding.rdoc [hunk 1]
```diff
@@ -274,6 +274,30 @@ For an \IO, \File, \ARGF, or \StringIO object, the internal encoding may be set
...
+  s = "R\u00E9sum\u00E9"
+  path = 't.tmp'
+  ext_enc = 'ISO-8859-1'
+  int_enc = 'UTF-8'
+
+  File.write(path, s, external_encoding: ext_enc)
+  raw_text = File.binread(path)
+
+  f = File.open(path, external_encoding: ext_enc, internal_encoding: int_enc)
+  transcoded_text = f.read
```
Comment (peterzhu2118) [inline doc/encoding.rdoc:291]: I think these two lines can be condensed into one line.

```suggestion
  transcoded_text = File.read(path, external_encoding: ext_enc, internal_encoding: int_enc)
```

**Review by peterzhu2118 — COMMENTED** [review]
One minor comment, otherwise looks good!

File: doc/encoding.rdoc (hunk 1 above)
Comment (nobu) [reply doc/encoding.rdoc:291]: Agree.
`open` without `close`/a block is not recommended.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**